File: 2023-12-18/leetcode_78.py
class Solution:
    def subsets(self, nums):
        answer = []

        def dfs(start = 0, subset = []):
            answer.append(subset)

            for idx in range(start, len(nums)):
                dfs(idx + 1, subset + [nums[idx]])
        dfs()
        return answer

File: leetcode/leetcode_78.js
const subsets = (nums) => {
  const answer = [];

  const dfs = (start = 0, subset = []) => {
    answer.push(subset);

    for (let idx = start; idx < nums.length; idx++) {
      dfs(idx + 1, subset.concat([nums[idx]]));
    }
  };
  dfs();
  return answer;
};
